pkg/api/types.go

```diff
@@ -315,6 +316,34 @@ type TestStep struct {
 	Resources   ResourceRequirements `json:"resources,omitempty"`
 }
 
+// TestStepCompact is the external representation of a test step that allows users to
+// write less verbose test configs
+type TestStepCompact struct {
```

Comment:
Usually we have the same name under two different packages. Can we move the internal one to `pkg/steps` or similar and then we won't need the `-Compact` suffix?